promql/parser/prettier.go

```diff
@@ -0,0 +1,543 @@
+				if items[forwardScanIndex].Typ == RIGHT_BRACE {
+					rightBraceIndex = forwardScanIndex
+					break
+				}
+			}
+			if leftBraceIndex == itemNotFound || rightBraceIndex == itemNotFound {
+				continue
+			}
+
+			itr = skipComments(items, itr)
+			labelNameindex := itr // index of __name__.
+			commaIndex := itr + 3
+
+			if items[commaIndex].Typ == COMMA {
+				skipBraces = rightBraceIndex-5 == skipComments(items, leftBraceIndex)
+			} else {
+				skipBraces = rightBraceIndex-4 == skipComments(items, leftBraceIndex)
+			}
+			identifierItem := Item{Typ: IDENTIFIER, Val: metricName, Pos: 0}
+			// The code below re-orders the lex items in the items array. The re-ordering of
+			// lex items must occur only when in a key-value pair, the key is `__name__`
+			// and the corresponding value is atomic.
+			//
+			// The leftBraceIndex and rightBraceIndex that form a range. Since the position of items beyond
+			// this range do not change, they are simply copied. However, the items within the range require
+			// re-ordering. So, they are added to the lex item slice after appending the identifier (metric_name)
+			// to the lex item slice.
+			//
+			// Also, we need to avoid `metric_name{}` case while re-ordering a `{__name__="metric_name"}`.
+			// Hence, if the indexes of left and right braces are next to each other, we skip
+			// appending them (braces) to the items slice.
+			tmp = append(tmp, items[:leftBraceIndex]...)
+			tmp = append(tmp, identifierItem)
+			if !skipBraces {
+				for j := leftBraceIndex; j <= rightBraceIndex; j++ {
+					if (items[commaIndex].Typ == COMMA && j == commaIndex) || (j >= labelNameindex && j < commaIndex) {
+						continue
+					}
+					tmp = append(tmp, items[j])
+				}
+			}
+			tmp = append(tmp, items[rightBraceIndex+1:]...)
+			items = tmp
+		case BY:
+			// Case: sum by() (metric_name) => sum(metric_name)
+			var (
+				leftParenIndex       = itemNotFound
+				rightParenIndex      = itemNotFound
+				groupingKeywordIndex = i
+				itr                  = skipComments(items, i)
+			)
+			for index := itr; index < len(items); index++ {
+				if items[index].Typ == LEFT_PAREN {
+					leftParenIndex = index
+				}
+				if items[index].Typ == RIGHT_PAREN {
+					rightParenIndex = index
+					break
+				}
+			}
+			if leftParenIndex+1 == rightParenIndex || leftParenIndex == rightParenIndex {
+				items = append(items[:groupingKeywordIndex], items[rightParenIndex+1:]...)
+			}
+		}
+	}
+	return p.refreshLexItems(items), nil
+}
+
+func (p *prettier) prettify(items []Item) (string, error) {
+	var (
+		result = ""
+		root   struct {
+			node      Node
+			hasScalar bool
+		}
+	)
+	for i, item := range items {
+		var (
+			nodeInfo       = &nodeInfo{head: p.Node, columnLimit: 100, item: item, items: items}
+			node           = nodeInfo.node()
+			nodeSplittable = nodeInfo.violatesColumnLimit()
+			// Binary expression use-case.
+			hasImmediateScalar               bool
```

Comment:
Else it's confusing what variables this comment refers to
```suggestion
			hasImmediateScalar               bool // For binary expression.
```